client/fisherman-service/src/handler.rs

```diff
@@ -0,0 +1,197 @@
+
+use crate::events::FishermanServiceEventBusProvider;
+
+pub(crate) const LOG_TARGET: &str = "fisherman-service";
+
+/// Commands that can be sent to the FishermanService actor
+#[derive(Debug)]
+pub enum FishermanServiceCommand {
+    /// Process a file deletion request by constructing proof of inclusion
+    /// from Bucket/BSP forest and submitting it to the blockchain
+    ProcessFileDeletionRequest {
+        /// The file key from the signed intention
+        signed_file_operation_intention: FileOperationIntention,
+        /// The signed intention
+        signature: MultiSignature,
+    },
+}
+
+/// Errors that can occur in the fisherman service
+#[derive(Error, Debug)]
+pub enum FishermanServiceError {
+    #[error("Database error: {0}")]
+    Database(#[from] diesel::result::Error),
+    #[error("Blockchain client error: {0}")]
+    Client(String),
+    #[error("Events retrieval error: {0}")]
+    EventsRetrieval(#[from] EventsRetrievalError),
+}
+
+/// The main FishermanService actor
+///
+/// This service monitors the StorageHub blockchain for file deletion requests,
+/// constructs proofs of inclusion from Bucket/BSP forests, and submits these proofs
+/// to the StorageHub protocol to permissionlessly mutate (delete the file key) the merkle forest on chain.
+pub struct FishermanService<RuntimeApi> {
+    /// Substrate client for blockchain interaction
+    client: Arc<ParachainClient<RuntimeApi>>,
+    /// Last processed block number to avoid reprocessing
+    last_processed_block: Option<BlockNumber>,
+    /// Event bus provider for emitting fisherman events
+    event_bus_provider: FishermanServiceEventBusProvider,
+}
+
+impl<RuntimeApi> FishermanService<RuntimeApi> {
+    /// Create a new FishermanService instance
+    pub fn new(client: Arc<ParachainClient<RuntimeApi>>) -> Self {
+        Self {
+            client,
+            last_processed_block: None,
+            event_bus_provider: FishermanServiceEventBusProvider::new(),
+        }
+    }
+
+    /// Monitor new blocks for file deletion request events
+    async fn monitor_block(
+        &mut self,
+        block_number: BlockNumber,
+        block_hash: H256,
+    ) -> Result<(), FishermanServiceError> {
+        debug!(target: LOG_TARGET, "🎣 Monitoring block #{}: {}", block_number, block_hash);
+
+        // TODO: Emit FileDeletionRequest event
+
+        self.last_processed_block = Some(block_number);
+        Ok(())
+    }
+}
+
+/// Implement the Actor trait for FishermanService
+impl<RuntimeApi> Actor for FishermanService<RuntimeApi>
+where
+    RuntimeApi: StorageEnableRuntimeApi + Send + 'static,
+    RuntimeApi::RuntimeApi: StorageEnableApiCollection + Send,
+{
+    type Message = FishermanServiceCommand;
+    type EventLoop = FishermanServiceEventLoop<RuntimeApi>;
+    type EventBusProvider = FishermanServiceEventBusProvider;
+
+    fn handle_message(
+        &mut self,
+        message: Self::Message,
+    ) -> impl std::future::Future<Output = ()> + Send {
```

Comment:
From clippy:
async fn handle_message(
```suggestion
    fn handle_message(
        &mut self,
        message: Self::Message,
    ) -> impl std::future::Future<Output = ()> + Send {
```
    &mut self,
    message: Self::Message,
)